### PR TITLE
[HomeTask3] | added javascript snippets for HomeTask3

### PR DESCRIPTION
**Pull request by bommas12**
@SaiTeja2904 please review the code snippets for hometask 3

https://kb.epam.com/display/GDOKB/FrontCamp+Batch-04+-+01+JAN+2021

partition: https://www.codewars.com/kata/partition-on
countwords: https://www.codewars.com/kata/word-count
calculateWithFunctions: https://www.codewars.com/kata/calculating-with-functions
getMiddle: https://www.codewars.com/kata/get-the-middle-character

### CODE REVIEW

File: HomeTask3/calculateWithFunctions.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/**
+ * 
+ * @param {number} num the number passed to return calculated value
+ * @param {function} func the function that needs to executed with num default value returns number
+ * 
+ * @throws {error} if the return number accepts number function again like zero
+ */
+const returnNumber = (num, func = (number) => number) => {
```
Comment (SaiTeja2904) [inline HomeTask3/calculateWithFunctions.js]: Don't use keywords such as number, string etc..

File: HomeTask3/calculateWithFunctions.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if (typeof func != "function") {
+        throw new Error("number can't process a number")
```
Comment (SaiTeja2904) [inline HomeTask3/calculateWithFunctions.js]: What is number can't process a number? Please rephrase it

File: HomeTask3/getMiddle.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Calculates and returns the middle part of string
+ * @param {string} str the string from which you want the middle characters
+ * 
+ * @returns {string} middle string
+ * @example getMiddle("test") returns "es", getMiddle("tests") returns "s"
+ */
+function getMiddle(str) {
+    let strlen = str.length;
```
Comment (SaiTeja2904) [inline HomeTask3/getMiddle.js]: Use const instead of let as you are not reassigning a new value to the variables

File: HomeTask3/partitionOn.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/**
+ * 
+ * @param {number} predicate the condition on which partioning should be done on
+ * @param {array} array the array that needs to be partitioned
+ * 
+ * @returns {number} boundary- the index of the second partition's first element
+ */
+function partitionOn(predicate, array) {
+    let truths = [], falses = [];
```
Comment (SaiTeja2904) [inline HomeTask3/partitionOn.js]: use const instead of let

File: HomeTask3/partitionOn.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    array.forEach(item => {
```
Comment (SaiTeja2904) [inline HomeTask3/partitionOn.js]: Please find another approach of solving this problem without forEach

**Review by SaiTeja2904 — CHANGES_REQUESTED** [review]

File: HomeTask3/calculateWithFunctions.js (hunk 1 above)
Comment (bommas12) [reply HomeTask3/calculateWithFunctions.js]: used num1,num2 instead

File: HomeTask3/calculateWithFunctions.js (hunk 2 above)
Comment (bommas12) [reply HomeTask3/calculateWithFunctions.js]: added "Operator not found between numbers" as Error message

File: HomeTask3/getMiddle.js (hunk 3 above)
Comment (bommas12) [reply HomeTask3/getMiddle.js]: done.

File: HomeTask3/partitionOn.js (hunk 5 above)
Comment (bommas12) [reply HomeTask3/partitionOn.js]: used array method reduce in js

File: HomeTask3/partitionOn.js (hunk 4 above)
Comment (bommas12) [reply HomeTask3/partitionOn.js]: done

File: HomeTask3/partitionOn.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * 
+ * @param {number} predicate the condition on which partioning should be done on
+ * @param {array} array the array that needs to be partitioned
+ * 
+ * @returns {number} boundary- the index of the second partition's first element
+ */
+function partitionOn(predicate, array) {
+    const result = array.reduce((obj, item) => {
+        obj[predicate(item) + "Array"].push(item);
```
Comment (SaiTeja2904) [inline HomeTask3/partitionOn.js]: Please change this. This is not readable

**Review by SaiTeja2904 — APPROVED** [review]

File: HomeTask3/partitionOn.js (hunk 6 above)
Comment (bommas12) [reply HomeTask3/partitionOn.js]: done using if else

**Pull request merged**